src/providers/activeLearning/objectDetection.ts

```diff
@@ -0,0 +1,197 @@
+import axios from "axios";
+import * as tf from "@tensorflow/tfjs";
+import { ElectronProxyHandler } from "./electronProxyHandler";
+import { LocalFileSystemProxy, ILocalFileSystemProxyOptions } from "../../providers/storage/localFileSystemProxy";
+
+// tslint:disable-next-line:interface-over-type-literal
+export type DetectedObject = {
+    bbox: [number, number, number, number];  // [x, y, width, height]
+    class: string;
+    score: number;
+};
+
+export type ImageObject = tf.Tensor3D|ImageData|HTMLImageElement|HTMLCanvasElement|HTMLVideoElement;
+
+export class ObjectDetection {
+    private modelLoaded: boolean = false;
+
+    get loaded(): boolean {
+        return this.modelLoaded;
+    }
+
+    private model: tf.GraphModel;
+    private jsonClasses: JSON;
+
+    /**
+     * Dispose the tensors allocated by the model. You should call this when you
+     * are done with the model.
+     */
+    public dispose() {
+        if (this.model) {
+            this.model.dispose();
+        }
+    }
+
+    public async load(modelFolderPath: string) {
+        try {
+            if (modelFolderPath.toLowerCase().startsWith("http://") ||
+                modelFolderPath.toLowerCase().startsWith("https://")) {
+                this.model = await tf.loadGraphModel(modelFolderPath + "/model.json");
+
+                const response = await axios.get(modelFolderPath + "/classes.json");
+                this.jsonClasses = JSON.parse(JSON.stringify(response.data));
+            } else {
+                if (modelFolderPath.toLowerCase().startsWith("file://")) {
+                    modelFolderPath = modelFolderPath.substring(7);
+                }
+                const handler = new ElectronProxyHandler(modelFolderPath);
+                this.model = await tf.loadGraphModel(handler);
+
+                const provider = new LocalFileSystemProxy();
+                this.jsonClasses = JSON.parse(await provider.readText(modelFolderPath + "/classes.json"));
+            }
+
+            // Warmup the model.
+            const result = await this.model.executeAsync(tf.zeros([1, 300, 300, 3])) as
```

Comment:
just curious about what 'warming up the model' - and the magic numbers in the array - actually do :slightly_smiling_face:

Reply:
:-)
The Tensor passed as input is a fake 300x300 image with 3 depth/colors.  The meaning for "warming up the model" is to fake a detection call in order to let the model load all necessary weights for his DL layers.